File: mycollab-services/src/main/java/com/esofthead/mycollab/module/crm/service/ibatis/TargetServiceImpl.java
/**
 * This file is part of mycollab-services.
 *
 * mycollab-services is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-services is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-services.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.module.crm.service.ibatis;

import com.esofthead.mycollab.core.persistence.ICrudGenericDAO;
import com.esofthead.mycollab.core.persistence.ISearchableDAO;
import com.esofthead.mycollab.core.persistence.service.DefaultService;
import com.esofthead.mycollab.module.crm.dao.CrmTaskMapper;
import com.esofthead.mycollab.module.crm.dao.TargetMapper;
import com.esofthead.mycollab.module.crm.dao.TargetMapperExt;
import com.esofthead.mycollab.module.crm.domain.SimpleTarget;
import com.esofthead.mycollab.module.crm.domain.Target;
import com.esofthead.mycollab.module.crm.domain.criteria.TargetSearchCriteria;
import com.esofthead.mycollab.module.crm.service.TargetService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional
public class TargetServiceImpl extends
        DefaultService<Integer, Target, TargetSearchCriteria> implements
        TargetService {

    @Autowired
    private TargetMapper targetMapper;

    @Autowired
    private TargetMapperExt targetMapperExt;

    @Autowired
    private CrmTaskMapper taskMapper;

    @Override
    public ICrudGenericDAO<Integer, Target> getCrudMapper() {
        return targetMapper;
    }

    @Override
    public ISearchableDAO<TargetSearchCriteria> getSearchMapper() {
        return targetMapperExt;
    }

    public SimpleTarget findTargetById(int targetId) {
        return targetMapperExt.findTargetById(targetId);
    }

}

File: mycollab-services/src/main/java/com/mycollab/module/crm/service/impl/TargetServiceImpl.kt
/**
 * Copyright © MyCollab
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.module.crm.service.impl

import com.mycollab.db.persistence.ICrudGenericDAO
import com.mycollab.db.persistence.ISearchableDAO
import com.mycollab.db.persistence.service.DefaultService
import com.mycollab.module.crm.dao.TargetMapper
import com.mycollab.module.crm.dao.TargetMapperExt
import com.mycollab.module.crm.domain.SimpleTarget
import com.mycollab.module.crm.domain.Target
import com.mycollab.module.crm.domain.criteria.TargetSearchCriteria
import com.mycollab.module.crm.service.TargetService
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional
class TargetServiceImpl(private val targetMapper: TargetMapper,
                        private val targetMapperExt: TargetMapperExt) : DefaultService<Int, Target, TargetSearchCriteria>(), TargetService {

    override val crudMapper: ICrudGenericDAO<Int, Target>
        get() = targetMapper as ICrudGenericDAO<Int, Target>

    override val searchMapper: ISearchableDAO<TargetSearchCriteria>
        get() = targetMapperExt

    override fun findTargetById(targetId: Int): SimpleTarget {
        return targetMapperExt.findTargetById(targetId)
    }

}
